ibc-core/ics04-channel/src/handler/send_packet.rs

```diff
@@ -30,6 +30,10 @@ pub fn send_packet_validate(
     ctx_a: &impl SendPacketValidationContext,
     packet: &Packet,
 ) -> Result<(), ContextError> {
+    if !packet.timeout_height_on_b.is_set() && !packet.timeout_timestamp_on_b.is_set() {
+        return Err(ContextError::PacketError(PacketError::MissingTimeout));
+    }
```

Comment:
shouldn't we do a similar check in recv, timeout and timeoutOnClose?
All the `on_a` vs `on_b` cause a lot of duplicate code, ideally this check should be done in one place for all messages that include a packet.

Reply:
Very good question! Here's my argument:
- OnRecvPacket: timeout doesn't matter - as this would generate packet_ack anyway.
- OnTimeout/OnTimeoutClose: we added `send_packet` with timeout height or timeout timestamp (invariant).

Also, I didn't find these checks in IBC spec and in ibc-go.